src/provider/default.ts

```diff
@@ -44,21 +42,15 @@ export class Provider implements ProviderInterface {
     if (providerOrOptions instanceof Provider) {
       // providerOrOptions is Provider
       this.provider = providerOrOptions.provider;
-    } else if (
-      providerOrOptions instanceof RpcProvider ||
-      providerOrOptions instanceof SequencerProvider
-    ) {
+    } else if (providerOrOptions instanceof RpcProvider) {
       // providerOrOptions is SequencerProvider or RpcProvider
       this.provider = <ProviderInterface>providerOrOptions;
     } else if (providerOrOptions && 'rpc' in providerOrOptions) {
       // providerOrOptions is rpc option
       this.provider = new RpcProvider(<RpcProviderOptions>providerOrOptions.rpc);
-    } else if (providerOrOptions && 'sequencer' in providerOrOptions) {
-      // providerOrOptions is sequencer option
-      this.provider = new SequencerProvider(<SequencerProviderOptions>providerOrOptions.sequencer);
     } else {
       // providerOrOptions is none, create SequencerProvider as default
-      this.provider = new SequencerProvider();
+      this.provider = new RpcProvider();
```

Comment:
Maybe change `ProviderOptions` to something like `type ProviderOptions = { rpc: RpcProviderOptions } | RpcProviderOptions;`, then the last `else if` can be removed and the `else` can have something like `new RpcProvider(providerOrOptions.rpc ?? providerOrOptions)`.

Reply:
In v6